src/memray/commands/attach.py

```diff
@@ -298,20 +299,21 @@ def run(self, args: argparse.Namespace, parser: argparse.ArgumentParser) -> None
         verbose = args.verbose
 
         if args.method == "auto":
-            if debugger_available("lldb", verbose=verbose):
-                args.method = "lldb"
-            elif debugger_available("gdb", verbose=verbose):
-                args.method = "gdb"
+            # Prefer gdb on Linux but lldb on macOS
+            if platform.system() == "Linux":
+                debuggers = ("gdb", "lldb")
+            else:
+                debuggers = ("lldb", "gdb")
+
+            for debugger in debuggers:
+                if debugger_available(debugger, verbose=verbose):
+                    args.method = debugger
+                    break
             else:
                 raise MemrayCommandError(
                     "Cannot find a supported lldb or gdb executable.",
                     exit_code=1,
                 )
-        elif not debugger_available(args.method, verbose=verbose):
-            raise MemrayCommandError(
-                f"Cannot find a supported {args.method} executable.",
-                exit_code=1,
-            )
```

Comment:
This changes the behavior of the non-auto mode. Before, if you specifically requested a debugger that wasn't available, you'd get:
```text
$ memray attach 24622 --method gdb --verbose
No gdb executable found
Cannot find a supported gdb executable.
```

Because of this block being removed, you now get a much less user-friendly message:
```text
Debugger command line: gdb -batch -p 24622 -nx -nw '-iex=set auto-solib-add off' '-ex=set $rtld_now=2' '-ex=set $libpath="/usr/local/lib/python3.11/site-packages/memray/_inject.abi3.so"' '-ex=set $port=34565' -x=/usr/local/lib/python3.11/site-packages/memray/commands/_attach.gdb
Traceback (most recent call last):
  File "/usr/local/bin/memray", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/local/lib/python3.11/site-packages/memray/commands/__init__.py", line 137, in main
    arg_values.entrypoint(arg_values, parser)
  File "/usr/local/lib/python3.11/site-packages/memray/commands/attach.py", line 343, in run
    errmsg = inject(args.method, args.pid, sidechannel_port, verbose=verbose)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/memray/commands/attach.py", line 105, in inject
    output = subprocess.check_output(cmd, text=True, stderr=subprocess.STDOUT)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/subprocess.py", line 466, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/subprocess.py", line 548, in run
    with Popen(*popenargs, **kwargs) as process:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/local/lib/python3.11/subprocess.py", line 1950, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'gdb'
```